kernel/src/net/socket/ip/datagram/mod.rs

```diff
@@ -352,6 +369,24 @@ impl Socket for DatagramSocket {
 
         Ok((copied_bytes, message_header))
     }
+
+    fn get_option(&self, option: &mut dyn SocketOption) -> Result<()> {
+        let options = self.options.read();
+
+        let res = options.socket.get_option(option);
+        match res {
```

Comment:
With the change suggested above, I think it is possible to write `match options.socket.get_option(option) {` directly.